cmd/zoekt-dynamic-indexserver/main.go

```diff
@@ -178,13 +200,22 @@ func (s *indexServer) serveTruncate(w http.ResponseWriter, r *http.Request) {
 	}
 	w.Header().Set("Content-Type", "application/json")
 	_ = json.NewEncoder(w).Encode(response)
+
+	incrementRequestsTotal(r, http.StatusOK)
+}
+
+func incrementRequestsTotal(r *http.Request, responseCode int) {
+	metricRequestsTotal.With(prometheus.Labels{"code": strconv.Itoa(responseCode), "method": r.Method, "path": r.URL.Path}).Inc()
```

Comment:
given `path` is under the user control, this can lead to a bad actor or a bug in the client blowing up your prometheus registry. Personally I'd only set hardcoded strings here. eg at each call site pass of incrementRequestsTotal pass in a route name.